integration-tests/src/test/java/com/example/IntegrationTest.java

```diff
@@ -848,11 +854,8 @@
     try {
       backend.create(ComponentWithDependencies.class);
       fail();
-    } catch (IllegalArgumentException e) {
-      assertThat(e).hasMessageThat()
-          .isEqualTo("com.example.ComponentWithDependencies "
-              + "declares dependencies [java.lang.String, java.lang.Runnable] "
-              + "and therefore must be created with a builder");
```

Comment:
This was a weird special-cased exception. Instead of duplicating for factory, just deleted and relying on other validation.